app/models/manageiq/providers/embedded_ansible/automation_manager/azure_credential.rb

```diff
@@ -47,7 +47,33 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::AzureCredential <
     :attributes => API_ATTRIBUTES
   }.freeze
 
+  alias secret auth_key
+
   def self.display_name(number = 1)
     n_('Credential (Microsoft Azure)', 'Credentials (Microsoft Azure)', number)
   end
+
+  def self.params_to_attributes(params)
+    attrs = params.reverse_merge(:options => {})
```

Comment:
I'm not sure what behavior we actually want here. Do we want options to be empty if none are provided or `nil`?

When migrating, I set it to `nil` if there are no options for a credential, even if we *could* have options for that type.

Reply:
I guess the problem with that is then we have to do extra existence checks when setting `params[:options][:my_val]` for the passed in `params[:options]`, or conditionally set `attrs[:options] = {}` if any of the expected keys exist.

Also, the helper methods (like `#client`, `#tenant`, and `#subscription`) become more complex, as we then need to do `nil` checks for each.

* * *

There are cons I am sure for either implementation, and I am not hard set on using what I have here, but that is the rational I used when putting what I have here together.

Let me know if you think we should change it.

Reply:
Well the column is nullable and we don't set a default value on the model so I don't think we can assume that it will always have a value regardless of what we do here.

Reply:
Fixed.